===================================================================
RCS file: /tcpdump/master/tcpdump/print-802_11.c,v
retrieving revision 1.42
retrieving revision 1.43
diff -u -r1.42 -r1.43
--- tcpdump/print-802_11.c	2006/06/13 22:25:30	1.42
+++ tcpdump/print-802_11.c	2007/02/01 02:18:18	1.43
@@ -264,7 +264,7 @@
 
 			if (pbody->tim.length <= 3)
 				break;
-			if (pbody->rates.length > sizeof pbody->tim.bitmap)
+			if (pbody->tim.length - 3 > sizeof pbody->tim.bitmap)
 				return;
 			if (!TTEST2(*(p + offset), pbody->tim.length - 3))
 				return;
